storage/innobase/fil/fil0crypt.cc

```diff
@@ -1117,9 +1125,22 @@ fil_crypt_space_needs_rotation(
 			break; // the space is already being processed and there are no more pages to rotate
 		}
 
-		crypt_data->rotate_state.scrubbing.is_active = false;
+		crypt_data->rotate_state.scrubbing.is_active = 
+			btr_scrub_start_space(space->id, &state->scrub_data);
+
+		time_t diff = time(0) - crypt_data->rotate_state.scrubbing.
+			last_scrub_completed;
+		btr_scrub_start_space(space->id, &state->scrub_data);
+
+		bool need_scrubbing =
+			(srv_background_scrub_data_uncompressed ||
+			 srv_background_scrub_data_compressed) &&
+			crypt_data->rotate_state.scrubbing.is_active
+			&& diff >= 0
+			&& ulint(diff) >= srv_background_scrub_data_interval;
+
 
-		if (need_key_rotation == false) {
+		if (need_key_rotation == false && need_scrubbing == false) {
```

Comment:
Later: if foo == false -> !foo (or why not (((foo == false) == true) != false) ...)